pynndescent/pynndescent_.py

```diff
@@ -171,9 +173,28 @@ def nn_descent(
             d = dist(data[i], data[indices[j]], *dist_args)
             heap_push(current_graph, i, d, indices[j], 1)
             heap_push(current_graph, indices[j], d, i, 1)
+            tried.add((i, indices[j]))
+            tried.add((indices[j], i))
 
     if rp_tree_init:
-        init_rp_tree(data, dist, dist_args, current_graph, leaf_array)
```

Comment:
I think this is just a conflict with changes @tomwhite made here; I think Tom's updated version is the way to go here as it it more generalisable.